lib/Onyx.js

```diff
@@ -1268,16 +1269,28 @@ function mergeCollection(collectionKey, collection) {
 /**
  * Insert API responses and lifecycle data into Onyx
  *
- * @param {Array} data An array of objects with shape {onyxMethod: oneOf('set', 'merge', 'mergeCollection'), key: string, value: *}
+ * @param {Array} data An array of objects with shape {onyxMethod: oneOf('set', 'merge', 'mergeCollection', 'multiSet', 'clear'), key: string, value: *}
  * @returns {Promise} resolves when all operations are complete
  */
 function update(data) {
     // First, validate the Onyx object is in the format we expect
-    _.each(data, ({onyxMethod, key}) => {
-        if (!_.contains([METHOD.CLEAR, METHOD.SET, METHOD.MERGE, METHOD.MERGE_COLLECTION], onyxMethod)) {
+    _.each(data, ({onyxMethod, key, value}) => {
+        if (!_.contains([METHOD.CLEAR, METHOD.SET, METHOD.MERGE, METHOD.MERGE_COLLECTION, METHOD.MULTI_SET], onyxMethod)) {
             throw new Error(`Invalid onyxMethod ${onyxMethod} in Onyx update.`);
         }
-        if (onyxMethod !== METHOD.CLEAR && !_.isString(key)) {
+        if (onyxMethod === METHOD.MULTI_SET) {
+            if (!_.isObject(value) || _.isArray(value) || _.isFunction(value)) {
+                throw new Error('Invalid value provided in Onyx multiSet. Onyx multiSet value must be of type object.');
+            }
+
+            _.each(_.keys(value), (multiSetKey) => {
+                if (_.isString(multiSetKey)) {
+                    return;
+                }
+
+                throw new Error(`Invalid ${typeof key} key provided in Onyx multiSet. Onyx key must be of type string.`);
+            });
```

Comment:
This seems not logically possible because all keys of an object must be strings?

Reply:
They can be other types as well. At least, what I'm looking to do if prevent number-type keys at the very minimum. You can see this in my unit test.: https://github.com/Expensify/react-native-onyx/pull/305/files#diff-fb75d1d7a8480f8429b43b89f94665322ee4576f95220815a5f7f3275e99fb40R605

Reply:
sorry, not trying to split hairs. I thought they would be cast to string. At least when iterating over them with `Object.keys()`, but maybe numbers are unique 🤷‍♂️ 

![2023-08-21_13-06-47](https://github.com/Expensify/react-native-onyx/assets/32969087/c46eb80a-500b-4be3-a2b4-a80866eab689)

NAB, I am not sure why should we not use a number as a key for Onyx. We would never do it. But doesn't seem like it should be explicitly prevented for any reason.

Reply:
Ah nice, okay. I can remove those checks then.